langserver/general.go

```diff
@@ -58,30 +58,14 @@ func (s *server) Initialize(_ context.Context, _ *protocol.ParamInitialize) (*pr
 
 // Initialized receives a confirmation by the client that the connection has been initialized
 // required by the protocol.Server interface.
-func (s *server) Initialized(ctx context.Context, _ *protocol.InitializedParams) (err error) {
+func (s *server) Initialized(_ context.Context, _ *protocol.InitializedParams) (err error) {
 	s.stateMu.Lock()
 	defer s.stateMu.Unlock()
 
 	if s.state != serverInitializing {
 		return errors.New("cannot initialize server: wrong server state")
 	}
 
-	if s.config.PrometheusURL != "" {
```

Comment:
Here normally it is safe to remove it because at anytime the `metadataService` is properly initialized and so this step is not anymore used.

But maybe I miss a step somewhere on how the server is initialized

Reply:
ok but if you look at the code, it will panic far before going here in case the metadataService cannot be initialized.

Basically it will fail directly in the method `ServerFromStream`.

Reply:
Before this PR, the `metadataService` would be initialized with an empty string as an URL in `serverFromStream`. Why should that panic?

Reply:
mmm yeah that's true. But then I have to store the `prometheusURL ` "just" for the initialization. It's a bit sad to keep an attribute for only one usage :'(.

Reply:
maybe one day I will find the good way to achieve it lol. Let's put it back then